client/test/unit/components/bhBreadCrumb.spec.js

```diff
@@ -0,0 +1,119 @@
+  var dropDownSpy = chai.spy();
+
+  var breadCrumb = {
+    path : [
+      { id : 'firstlink', label: 'path1', link: '#/path1_link_to_go' },
+      { label: 'path2', link: '#/path2_link_to_go' },
+      { label: 'path3', link: '#/path3_link_to_go', current: true }
+    ],
+
+    button : [
+      { id : 'printbtn', icon: 'glyphicon glyphicon-print', label: 'Print', action: buttonSpy },
+      { id : 'repeatbtn', icon: 'glyphicon glyphicon-repeat', label: 'Repeat', action: buttonSpy, color: 'btn-danger' },
+      { id : 'refreshbtn', icon: 'glyphicon glyphicon-refresh', label: 'Refresh', action: buttonSpy}
+    ],
+
+    label : [
+      { icon: 'glyphicon glyphicon-print', label: 'My Label 1' },
+      { label: 'My label 2' },
+      { label: 'My label 3' }
+    ],
+
+    dropdown : [
+      {
+        id : 'dropdownone',
+        label : 'Dropdown 1',
+        color : 'btn-primary',
+        option : [
+          { id : 'optionFranc', label : 'Fc', action : dropDownSpy },
+          { id : 'optionDollar', label : '$', action : dropDownSpy }
+        ]
+      },
+
+      {
+        id : 'dropdowntwo',
+        label : 'Dropdown 2',
+        color : 'btn-success',
+        option : [
+          { id : 'long', label : 'item1 dd2 with a too long text that you can imagine', action : dropDownSpy },
+          { id : 'short', label : 'item2 dd2', action : dropDownSpy}
+        ]
+      }
+    ]
+  }, component;
+
+  var element, scope, $compile;
+
+  var template = `
+    <bh-breadcrumb
+      path="config.path"
+      button="config.button",
+      label="config.label",
+      dropdown="config.dropdown">
+    </bh-breadcrumb>`;
+
+  beforeEach(module('pascalprecht.translate', 'ngStorage', 'angularMoment', 'bhima.services', 'bhima.components', 'templates'));
+
+  beforeEach(inject(function (_$rootScope_, _$compile_, _$componentController_) {
+      scope = _$rootScope_.$new();
+      $compile = _$compile_;
+      component = _$componentController_('bhBreadcrumb', null, {
+        label : breadCrumb.label,
+        button : breadCrumb.button,
+        dropdown : breadCrumb.dropdown,
+        path : breadCrumb.path
+      });
+  }));
+
+  it('receives path data correctly', function (){
+    expect(component.path.length).to.be.equal(3);
+  });
+
+  it('receives button data correctly', function (){
+    expect(component.button.length).to.be.equal(3);
+  });
+
+  it('receives label data correctly', function (){
+    expect(component.label.length).to.be.equal(3);
+  });
+
+  it('receives dropdown data correctly', function (){
+    expect(component.dropdown.length).to.be.equal(2);
+  });
+
+  it('fires methods passed in through the button parameters', function () {
+
+    // assign configuration object to scope and init it
+    scope.config = breadCrumb;
+    element = $compile(angular.element(template))(scope);
+    scope.$digest();
+
+    scope.config.button.forEach(function (btn) {
+      $(element).find('#' + btn.id).click();
```

Comment:
If we ever do remove jQuery as a dependency (#176), tests depending on it will break.  It turns out that `angular.element()` is [an alias for the jQuery function](https://docs.angularjs.org/api/ng/function/angular.element) that will default to a smaller implementation if jQuery is not available.  I would advise that we use `angular.element()` instead of jQuery going forward.

From the documentation:
> If jQuery is available, angular.element is an alias for the jQuery function. If jQuery is not available, angular.element delegates to Angular's built-in subset of jQuery, called "jQuery lite" or jqLite.